agenta-web/src/components/Evaluations/EvaluationErrorModal.tsx

```diff
@@ -1,15 +1,30 @@
-import React from "react"
+import React, {Dispatch, SetStateAction} from "react"
 import {Modal, Button} from "antd"
 
 interface Props {
     isModalOpen: boolean
-    onClose: () => void
     handleNavigate: () => void
     message: string
+    btnText: string
+    closeBtnText: Dispatch<SetStateAction<string>>
+    closeEvalutionError: Dispatch<SetStateAction<string>>
+    closeEndpoint: Dispatch<SetStateAction<string>>
 }
 
-const EvaluationErrorModal: React.FC<Props> = ({isModalOpen, onClose, handleNavigate, message}) => {
-    const handleCloseModal = () => onClose()
+const EvaluationErrorModal: React.FC<Props> = ({
+    isModalOpen,
+    handleNavigate,
+    message,
+    btnText,
+    closeBtnText,
+    closeEvalutionError,
+    closeEndpoint,
+}) => {
+    const handleCloseModal = () => {
```

Comment:
I would keep it like before:

`const handleCloseModal = () => onClose()`

Instead, call the setter function inside the callback from the parent component.